File: ObjectDT/binary.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

using namespace cv;
using namespace std;

int main(int argc, char **argv){
    if(argc < 2){
        cerr <<"Send image"<< endl;
        return EXIT_FAILURE;
    }

    Mat image = imread(argv[1]);
    if(image.empty()){
        cerr <<"Fail to load image"<< endl;
        return EXIT_FAILURE;
    }

    Mat gray;
    cvtColor(image, gray, COLOR_BGR2GRAY);

    Mat binary;
    threshold(gray, binary, 127, 255, THRESH_BINARY);

    imshow("Original Image", image);
    imshow("Binary Image", binary);
    imwrite("binary.jpg", binary);

    waitKey(0);
    destroyAllWindows();
    return EXIT_SUCCESS;
}

File: ObjectDT/binary.py
import cv2 
import numpy as np

path = "../Data/Eye.jpg"
img = cv2.imread(path)
gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)

#binary
thresh = 80
max = 255
_, thre = cv2.threshold(gray, thresh, max, cv2.THRESH_BINARY)

cv2.imshow("Binary Image", thre)
cv2.waitKey(0)
cv2.destroyAllWindows()